### ISSUE TITLE
Unit 7 Lesson 46

### ISSUE THREAD

**Issue by mentat31** [DanielaSfregola/get-programming-with-scala#97]
Trying to do the Database Queries with Quill lesson and seem to be running into a problem. 

When calling queries.testConnection() I get the error; 

`com.github.jasync.sql.db.exceptions.UnsupportedAuthenticationMethodException: Unknown authentication method -> '10'`

I've tried looking into the container with little luck by finding some helpful messages about modifying the postgresql.config before the container starts... but this is docker, so before making my own image I just wanted to see if anyone else has had this problem. On macOS if that helps.

**Comment by DanielaSfregola** [comment]
Hi @mentat31,
have you tried running the solution (code published in this repo)? That will tell us if your problem is code or setup related.

Cheers,
Daniela

**Comment by lamazoni** [comment]
I m experiencing this pb too. If I put a breakpoint I can see that I can connect to the db with a tool like omnidb with a string like postgresql://localhost:32805/test?user=test&password=test
but when modify the following seems to have no effect :
PostgreSQL.scala (unit7 lesson 47)
```scala
    val components = List(
//      container.getJdbcUrl,
      "postgresql://" + container.getContainerIpAddress + ":" + container.getMappedPort(PostgreSQLContainer.POSTGRESQL_PORT) + "/" + container.getDatabaseName,
      s"user=${container.getUsername}",
      s"password=${container.getPassword}"
    ) 
```
I was thinking that there was a auth parameter somewhere (I had some sucess by launching manually a container with -e POSTGRES_INITDB_ARGS=–auth-host=md5) but I can find any ...

**Comment by mentat31** [comment]
> Hi @mentat31, have you tried running the solution (code published in this repo)? That will tell us if your problem is code or setup related.
> 
> Cheers, Daniela

Just seeing this today, however after a couple of days of debugging I found a solution. I noticed in the log that AuthenticationStartupParser.kt didn't have all the authentication methods that I was seeing on their github. 

I added `"com.github.jasync-sql" % "jasync-postgresql" % "2.0.7"` to my build.sbt and it now seems to be working! Thank you for your response!
